vowpalwabbit/v_array.h

```diff
@@ -429,24 +429,18 @@ void copy_array(v_array<T>& dst, const v_array<T>& src, T (*copy_item)(const T&)
 }
 
 template <class T>
+VW_DEPRECATED("Use v_array::insert instead")
 void push_many(v_array<T>& v, const T* src, size_t num)
 {
-  if (v._end + num >= v.end_array)
-    v.resize(std::max(2 * (size_t)(v.end_array - v._begin) + 3, v._end - v._begin + num));
-#ifdef _WIN32
-  memcpy_s(v._end, (v.end_array - v._end) * sizeof(T), src, num * sizeof(T));
-#else
-  memcpy(v._end, src, num * sizeof(T));
-#endif
-  v._end += num;
+  v.insert(v.end(), src, src + num);
 }
 
 template <class T>
+VW_DEPRECATED("calloc_reserve is no longer supported. You should use appropriate constructors instead.")
 void calloc_reserve(v_array<T>& v, size_t length)
 {
-  v._begin = calloc_or_throw<T>(length);
-  v._end = v._begin;
-  v.end_array = v._begin + length;
+  v.reserve(length);
```

Comment:
We should call clear() before we reserve

Reply:
Prior to this change it would leak if there were anything inside so this is kind of relying on a user passing in a default v_array. That being said I see no harm in clearing